addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -546,6 +548,59 @@ function factory(dependencies) {
             }
         }
 
+        /**
+         * @private
+         * @param {Object} param0
+         * @param {number} param0.last_seen_message_id
+         * @param {number} param0.partner_id
+         * @param {number} param0.channel_id
+         * @param {number} param0.channel_unread_counter
+         */
+        _handleNotificationMessageUnread({ last_seen_message_id, partner_id, channel_id, channel_unread_counter }) {
+            const channel = this.env.models['mail.thread'].findFromIdentifyingData({
+                id: channel_id,
+                model: 'mail.channel',
+            });
+            if (!channel) {
+                // for example seen from another browser, the current one has no
+                // knowledge of the channel
+                return;
+            }
+            let lastMessage;
+            if (last_seen_message_id) {
+                lastMessage = this.env.models['mail.message'].insert({ id: last_seen_message_id });
+            }
+            // restrict computation of seen indicator for "non-channel" channels
+            // for performance reasons
+            const shouldComputeSeenIndicators = channel.channel_type !== 'channel';
+            if (shouldComputeSeenIndicators) {
+                this.env.models['mail.thread_partner_seen_info'].insert({
+                    channelId: channel.id,
+                    lastSeenMessage: lastMessage ? link(lastMessage) : unlink(),
+                    partnerId: partner_id,
+                });
+                if (lastMessage) {
+                    this.env.models['mail.message_seen_indicator'].insert({
+                        channelId: channel.id,
+                        messageId: lastMessage.id,
+                    });
+                }
+            }
+            if (this.messaging.currentPartner.id === partner_id) {
+                channel.update({
+                    lastSeenByCurrentPartnerMessageId: last_seen_message_id,
+                    pendingSeenMessageId: undefined,
+                    serverMessageUnreadCounter: channel_unread_counter,
+                });
+            }
+            if (shouldComputeSeenIndicators) {
+                // FIXME force the computing of thread values (cf task-2261221)
+                this.env.models['mail.thread'].computeLastCurrentPartnerMessageSeenByEveryone(channel);
+                // FIXME force the computing of message values (cf task-2261221)
+                this.env.models['mail.message_seen_indicator'].recomputeSeenValues(channel);
+            }
```

Comment:
I think all of this could be finally done with proper compute in models? It was hard before because we had difficulties to express dependencies but it's not longer needed, and we never bothered to clean it.

But I don't want use to duplicate that code  here.

Reply:
Is this specific to those FIXMEs or are you suggesting that most of what this function does should be implemented through computes?